support/cas-server-support-x509-webflow/src/main/java/org/apereo/cas/web/flow/X509CertificateCredentialsNonInteractiveAction.java

```diff
@@ -21,21 +25,41 @@
  */
 @Slf4j
 public class X509CertificateCredentialsNonInteractiveAction extends AbstractNonInteractiveCredentialsAction {
-
     
     private static final String CERTIFICATE_REQUEST_ATTRIBUTE = "javax.servlet.request.X509Certificate";
 
+    private boolean extractCertificateFromRequest;
+
+    @Autowired(required = false)
+    private ExtractX509Certificate x509ExtractSSLCertificate;
```

Comment:
This is generally not a good idea. It would be better if you create a NoOp instead, or create a new Action class and inject it dynamically when settings allow for it.

Reply:
I haven't addressed this yet. What about this is not a good idea? Autowiring a an optional bean into an Action? When you suggest making a new Action, do you mean have x509check() method return a different Action depending on the feature being enabled?

Thanks.

Reply:
Through painful practice, we have learned that introducing an auto-wired field into a class that is somehow somewhere expected and assumed to turn into a Bean causes all sorts of weird issues, least of all for tests. There are a number of lingering auto-wired fields still throughout the codebase, all of which should and are getting consistently cleaned up where the bit is moved to the appropriate `Configuration` class.

As for the suggestion; yes that's sort of what I had in mind. You have the default action today in CAS, and your particular behavior would be an extended version of the default with some additional ops to extract certificates. If the child action fails, simply delegate back to the default. Then as you note, the actual bean definition would be constructed based on the presence of a particular setting. 

If you need to chain the two actions together and have them, one at a time, process a request...we can discuss that too and there are examples throughout that sort of show how to do composite type operations without introducing the risk or check for NPEs.